tutorials/introductory/customizing.py

```diff
@@ -130,52 +132,106 @@
+#
+# Every time when we *import* matplotlib, it will look for :file:`matplotlibrc`
+# in four locations, in the following order:
 #
 # 1. :file:`matplotlibrc` in the current working directory, usually used for
 #    specific customizations that you do not want to apply elsewhere.
 #
-# 2. :file:`$MATPLOTLIBRC` if it is a file, else :file:`$MATPLOTLIBRC/matplotlibrc`.
+# 2. When environment variable ``$MATPLOTLIBRC`` exists, matplotlib then looks
+#    for :file:`$MATPLOTLIBRC` if it is a file, else :file:`$MATPLOTLIBRC/matplotlibrc`.
 #
 # 3. It next looks in a user-specific place, depending on your platform:
 #
-#    - On Linux and FreeBSD, it looks in :file:`.config/matplotlib/matplotlibrc`
-#      (or `$XDG_CONFIG_HOME/matplotlib/matplotlibrc`) if you've customized
-#      your environment.
+#    - On Linux and FreeBSD, it looks in :file:`$HOME/.config/matplotlib/matplotlibrc`
+#      or :file:`$XDG_CONFIG_HOME/matplotlib/matplotlibrc` if the environment
+#      variable ``$XDG_CONFIG_HOME`` is set.
 #
-#    - On other platforms, it looks in :file:`.matplotlib/matplotlibrc`.
+#    - On other platforms, it looks in :file:`$HOME/.matplotlib/matplotlibrc`.
 #
-#    See :ref:`locating-matplotlib-config-dir`.
+#    See :ref:`locating-matplotlib-config-dir` for more details on how to find
+#    out these locations.
 #
-# 4. :file:`{INSTALL}/matplotlib/mpl-data/matplotlibrc`, where
-#    :file:`{INSTALL}` is something like
-#    :file:`/usr/lib/python3.7/site-packages` on Linux, and maybe
-#    :file:`C:\\Python37\\Lib\\site-packages` on Windows. Every time you
-#    install matplotlib, this file will be overwritten, so if you want
-#    your customizations to be saved, please move this file to your
+# 4. :file:`{PYINSTALL}/site-packages/matplotlib/mpl-data/matplotlibrc`. Every
+#    time you install/update matplotlib, this file will be overwritten, so if
+#    you want your customizations to be saved, please move this file to your
 #    user-specific matplotlib directory.
 #
 # Once a :file:`matplotlibrc` file has been found, it will *not* search any of
-# the other paths.
+# the other locations.
 #
 # To display where the currently active :file:`matplotlibrc` file was
 # loaded from, one can do the following::
 #
-#   >>> import matplotlib
-#   >>> matplotlib.matplotlib_fname()
+#   >>> import matplotlib as mpl
+#   >>> mpl.matplotlib_fname()
 #   '/home/foo/.config/matplotlib/matplotlibrc'
 #
-# See below for a sample :ref:`matplotlibrc file<matplotlibrc-sample>`.
+#
+# Formats and styling of :file:`matplotlibrc` file
+# ------------------------------------------------
+#
+# The :file:`matplotlibrc` file is best viewed in a editor which supports Python
+# mode syntax highlighting. Blank lines, lines starting with a comment symbol
+# are ignored, as are trailing comments.  Other lines must have the format::
+#
+#     key : val  ## optional comment
+#
+#
+# The :file:`matplotlibrc` file follows these relaxed style conventions:
```

Comment:
I would not bother with documenting the *layout style* of the matplotlibrc in the introductory tutorial.  Actually I wouldn't bother documenting the style of matplotlibrc at all -- basically the only thing that matters is that matplotlibrc.template is internally consistent (I don't think the styles shipped in mpl-data/stylelib are very much read...) and its style is pretty obvious.  (Perhaps it (barely) made sense to have that in matplotlibrc.template, but not here.)

Reply:
Thank you very much for the thoughts! I agreed that put this styling statement in this introductory tutorial indeed doesn't make much sense.

And as you said, internally consistent do matters for the `matplotlibrc.template` file. But previously, it's not that consistent from part to part in `matplotlibrc.template`, and this styling statement was trying to strengthen that awareness.

So I think it could be helpful for this layout style appears somewhere, and at this time, I put this styling statement back to the `matplotlibrc.template` file. And as always, I am happy to hear and learn from any suggestion!